misoc/integration/builder.py

```diff
@@ -165,11 +165,19 @@ def _initialize_rom(self):
         if self.soc.integrated_rom_size:
             with open(bios_file, "rb") as boot_file:
                 boot_data = []
+                unpackEndian = ">I" if self.soc.cpu_type != "vexriscv" else "<I"
                 while True:
                     w = boot_file.read(4)
                     if not w:
                         break
-                    boot_data.append(struct.unpack(">I", w)[0])
+                    boot_data.append(struct.unpack(unpackEndian, w)[0])
+
+            if self.soc.cpu_type == "vexriscv":
```

Comment:
You've already swapped the endianness above, no need to do it again.

Reply:
You mean the "rom CRC endian reswap" is useless ?
Because in simulation, without that reswap, the CRC check fail.

Reply:
Why are you swapping everything twice for vexriscv? There is some bug here. I don't know where but it exists.

Reply:
I can only see one swap, for a single 32 bit word (the one of the CRC). 

The first stuff with the struct.unpack isn't there to change the endianess, but only to decode the file and generate the rom words.
FileBytes[0] -> rom[0][7:0]
FileBytes[1] -> rom[0][15:8]
FileBytes[2] -> rom[0][24:16]
FileBytes[3] -> rom[0][31:24]
FileBytes[4] -> rom[1][7:0]

n.b. RISC-V is little endian, LM32 is big endian

Reply:
Oh sorry, I misread.

Reply:
The thing which trouble me the most, is that between the LM32 and the VexRiscv (Big vs Little endian) i had nothing to change to have the serial peripheral working ^.^

Apparently both agree that LSB is bit zero and MSB is bit 31 (on the 32 bit whishbone bus).

Reply:
Yes? what is wrong with that?